src/arch/southern-islands/driver/DriverCall.cc

```diff
@@ -902,6 +902,13 @@ int Driver::CallNDRangeFree(comm::Context *context,
 		throw Error(misc::fmt("%s: invalid ndrange ID (%d)", 
 			__FUNCTION__, ndrange_id));
 
+	// Unmap NDRange if it is timing
```

Comment:
- Title for declaration of `ndrange_id`, or group under `// Read arguments` section.
- Write full sentences in comments (`Unmap ND-Range if we are running a timing simulation`)
- I'm not sure if this is the place where we should be unmapping the ND-Range. First, it is not a good idea, unless strictly necessary, to create a reverse dependency between the emulator (or driver) and the timing simulator. Second, do we really need to fully unmap it based on an action of the driver? Isn't it possible to just unmap it when there is no uop remaining from any wavefront from any work-group in the ND-Range? This is an event that can be detected directly in the timing simulator, and not in the driver.